libtiledbsoma/src/soma/soma_array.cc

```diff
@@ -123,11 +123,11 @@ SOMAArray::SOMAArray(
 }
 
 void SOMAArray::open(tiledb_query_type_t query_type) {
-    mq_->array().get()->open(query_type);
+    arr_->open(query_type);
```

Comment:
Nice.   And because you have a private `arr_` I think we do not need the exposed `array()` function you added and keep the interface more opaque:

https://github.com/single-cell-data/TileDB-SOMA/blob/4a496e51acb5320be2c6568bd54b235ef99c0114/libtiledbsoma/src/soma/managed_query.h#L273